test/docfx.Test/DocfxTest.cs

```diff
@@ -206,30 +235,67 @@ private static async Task RunLanguageServer(string docsetPath, DocfxTestSpec spe
+            var randomContinueOutputPath = Path.ChangeExtension(outputPath, $".continue.{Guid.NewGuid()}");
 
             docsetPath = Path.Combine(docsetPath, spec.Cwd ?? "");
 
+            var isOutputTypePageJson = IsOutputTypePageJson(spec);
+            var isTwoStagesBuild = isOutputTypePageJson && !dryRun && !spec.NoContinue;
             using (TestUtility.EnsureFilesNotChanged(docsetPath, spec.NoInputCheck))
             {
                 if (singleFile)
                 {
-                    RunSingleFileBuild(docsetPath, randomOutputPath, spec, package);
+                    RunSingleFileBuild(docsetPath, randomNormalOutputPath, spec, package);
                 }
                 else
                 {
                     var commandLine = new[]
                     {
                         "build", docsetPath,
-                        "--output", randomOutputPath,
-                        "--log", Path.Combine(randomOutputPath, ".errors.log"),
+                        "--output", randomNormalOutputPath,
+                        "--log", Path.Combine(randomNormalOutputPath, ".errors.log"),
                         dryRun ? "--dry-run" : null,
                         spec.NoRestore ? "--no-restore" : null,
                         spec.NoCache ? "--no-cache" : null,
                         spec.NoDrySync ? "--no-dry-sync" : null,
                     }.Concat(spec.BuildFiles.SelectMany(file => new[] { "--file", Path.Combine(docsetPath, file) }));
 
                     Docfx.Run(commandLine.Where(arg => arg != null).ToArray(), package);
+
+                    if (isTwoStagesBuild)
+                    {
+                        var firstStageCommandLine = new[]
+                        {
+                            "build", docsetPath,
+                            "--output", randomFirstStageOutputPath,
+                            "--log", Path.Combine(randomFirstStageOutputPath, ".errors.log"),
+                            "--output-type", "json",
+                            spec.NoRestore ? "--no-restore" : null,
+                            spec.NoCache ? "--no-cache" : null,
+                            spec.NoDrySync ? "--no-dry-sync" : null,
+                        };
+
+                        Docfx.Run(firstStageCommandLine.Where(arg => arg != null).ToArray(), package);
+
+                        RemoveUnnecessaryFilesForContinue(randomFirstStageOutputPath);
+
+                        var templatePath = GetTemplatePath(package, s_repos.Value);
+                        var continueCommandLine = new[]
+                        {
+                            "build", randomFirstStageOutputPath,
+                            "--output", randomContinueOutputPath,
+                            "--log", Path.Combine(randomContinueOutputPath, ".errors.log"),
+                            "--continue",
+                            "--template", templatePath,
+                            spec.NoRestore ? "--no-restore" : null,
```

Comment:
Consider explicitly pass `--output-type pageJson`. We should also check that `--continue` must be paired with `--output-type pagejson` in the build command, because output types like HTML is not supported yet.